pkg/plugins/policies/core/xds/endpoints.go

```diff
@@ -38,6 +39,30 @@ func GatherEndpoints(rs *xds.ResourceSet) EndpointMap {
 	return em
 }
 
+func GatherGatewayEndpoints(rs *xds.ResourceSet) EndpointMap {
+	em := EndpointMap{}
+	for _, res := range rs.Resources(envoy_resource.EndpointType) {
```

Comment:
maybe we can make method `gatherEndpoints` which has parameter `gatherEndpoints(rs *xds.ResourceSet, origin string)`